modules/tracking/samples/lasot_benchmark.py

```diff
@@ -0,0 +1,246 @@
+import numpy as np
+import cv2 as cv
+import argparse
+import os
+
+
+def calc_coords(bb):
+    '''
+    The method used to evaluate coordinates of corners of the bounding box.
+    Input: tuple (x, y, w, h), where:
+    x - "x" coordinates of the top left corner
+    y - "y" coordinates of the top left corner
+    w - width of the bounding box
+    h - the height of the bounding box
+    Output: dictionary coordinates contain names of the values as keys and
+    numerical values as values
+    '''
+    xmin = bb[0]
+    xmax = bb[0] + bb[2] - 1.0
+    ymin = bb[1]
+    ymax = bb[1] + bb[3] - 1.0
+    cx = bb[0] + (bb[2] + 1.0) / 2
+    cy = bb[1] + (bb[3] + 1.0) / 2
+    coords = {'xmin': xmin, 'xmax': xmax, 'ymin': ymin, 'ymax': ymax,
+              'cx': cx, 'cy': cy}
+    return coords
+
+
+def get_iou(new, gt):
+    '''
+    During the calculation of intersection over union, we are checking
+    numerical value of area_of_overlap, because if it is equal to 0,
+    we have no intersection.
+    Lists, generated by 'calc_coords', are used as input for these methods.
+
+    Respectively:
+    new[0],gt[0] -> xmin, new[1],gt[1] -> xmax, new[2],gt[2] -> ymin
+    new[3],gt[3] -> ymax, new[4],gt[4] -> cx, new[5],gt[5] -> cy
+    '''
+    new_xmin, new_xmax, new_ymin, new_ymax, new_cx, new_cy = new
+    gt_xmin, gt_xmax, gt_ymin, gt_ymax, gt_cx, gt_cy = gt
+    dx = max(0, min(new_xmax, gt_xmax) - max(new_xmin, gt_xmin))
+    dy = max(0, min(new_ymax, gt_ymax) - max(new_ymin, gt_ymin))
+    area_of_overlap = dx * dy
+    if area_of_overlap != 0:
+        area_of_union = (new_xmax - new_xmin) * (new_ymax - new_ymin) + (
+            gt_xmax - gt_xmin) * (gt_ymax - gt_ymin) - area_of_overlap
+        iou = area_of_overlap / area_of_union
+    else:
+        iou = 0
+    return iou
+
+
+def get_pr(new, gt):
+    '''
+    In calculations of precision and normalized precision are used thresholds
+    from original TrackingNet paper: for metric calculation TrackingNet using
+    lists of thresholds, but here we use only one numerical value of the
+    threshold for each metric
+    Lists, generated by 'calc_coords', are used as input for these methods.
+
+    Respectively:
+    new[0],gt[0] -> xmin, new[1],gt[1] -> xmax, new[2],gt[2] -> ymin
+    new[3],gt[3] -> ymax, new[4],gt[4] -> cx, new[5],gt[5] -> cy
+    '''
+    new_xmin, new_xmax, new_ymin, new_ymax, new_cx, new_cy = new
+    gt_xmin, gt_xmax, gt_ymin, gt_ymax, gt_cx, gt_cy = gt
+    precision = np.sqrt((new_cx - gt_cx) ** 2 + (new_cy - gt_cy) ** 2)
+    return precision
+
+
+def get_norm_pr(new, gt, gt_bb_w, gt_bb_h):
```

Comment:
We can combine `get_norm_pr` with `get_pr`

Reply:
I do not think it is necessary, because it is different metrics and for flexibility of the benchmark it is better if we can work with every metric independently.

Reply:
These metrics are very similar and easy to combine

Reply:
```python
def get_precision(bbox, gt, is_norm=False):
    '''
    In calculations of precision and normalized precision are used thresholds
    from original TrackingNet paper: for metric calculation TrackingNet using
    lists of thresholds, but here we use only one numerical value of the
    threshold for each metric
    '''
    bb_x, bb_y, bb_w, bb_h = bbox
    gt_x, gt_y, gt_w, gt_h = gt
    def get_center(coord, size): return coord + (size + 1.0) / 2
    bb_cx, bb_cy = get_center(bb_x, bb_w), get_center(bb_y, bb_h)
    gt_cx, gt_cy = get_center(gt_x, gt_w), get_center(gt_y, gt_h)
    if is_norm:
        bb_cx, gt_cx = bb_cx / gt_w, gt_cx / gt_w
        bb_cy, gt_cy = bb_cy / gt_h, gt_cy / gt_h

    thr = 0.1 if is_norm else 20
    normalized_precision = np.sqrt((bb_cx - gt_cx)**2 + (bb_cy - gt_cy)**2)
    norm_pr_value = 0.0 if normalized_precision > thr else 1.0
    return norm_pr_value
```

Reply:
Yes, but if user want to remove one of them, just comment it is enough. If we combine these methods, user would be forced to change method inside it. It is not conveniently.

Reply:
I don't understand you. If user wants to get only normalized precision, he can:
`norm_pr = get_precision(bbox, gt, is_norm=True)`
or 
`pr  = get_precision(bbox, gt, is_norm=False)`

Why might the user need to remove function?

Reply:
It is about flexibility of the benchmark: I want to make the benchmark flexible for user.
If user want to add new metrics or remove old ones it is more convenient to operate these metrics via methods. Combining metrics is good for now, but it is not convenient for the future work, improvements and changes.

Reply:
The user can simply add new metrics rather than remove old ones

Reply:
But I still think that 1 metric  == 1 method is more convenient.

Reply:
Code duplication is not very good practice. These two metrics are actually similar so it's better to have a single implementation.

Reply:
We are talking about few lines of code - it is partially duplicated code, but it is better for scalability of the benchmark. I do not see anything bad about these functions.

Reply:
It’s more clear for the user to use one function with the `is_norm` flag for similar metrics than two different functions with a minimum difference

Reply:
@l-bat, why not
```python
def get_pr(new, gt, is_norm):
    new_xmin, new_xmax, new_ymin, new_ymax, new_cx, new_cy = new
    gt_xmin, gt_xmax, gt_ymin, gt_ymax, gt_cx, gt_cy = gt
    dx = new_cx - gt_cx
    dy = new_cy - gt_cy
    if is_norm:
       dx /= gt_bb_w
       dy /= gt_bb_h

    return np.sqrt(dx ** 2 + dy ** 2)
```

Reply:
It seems to me that `calc_coords` isn't needed and we can calculate `cx, cy` here. It’s a good idea to normalize only the difference.